src/olympia/activity/admin.py

```diff
@@ -31,4 +32,52 @@ def has_delete_permission(self, request, obj=None):
         return False
 
 
+class ReviewActionReasonLogAdmin(admin.ModelAdmin):
+    date_hierarchy = 'created'
+    fields = (
+        'created',
+        'activity_log',
+        'user_email',
+        'reason',
+    )
+    list_display = (
+        'created',
+        'activity_log',
+        'reason',
+        'user_email',
+    )
+    list_filter = ('reason',)
+    readonly_fields = (
+        'created',
+        'activity_log',
+        'user_email',
+    )
+    search_fields = ('activity_log__user__email',)
+    view_on_site = False
+
+    @admin.display()
+    def user_email(self, obj):
+        return obj.activity_log.user.email
+
+    def has_add_permission(self, request):
+        return False
+
+    def has_delete_permission(self, request, obj=None):
+        return False
+
+    def formfield_for_foreignkey(self, db_field, request, **kwargs):
+        if db_field.name == 'reason':
+            kwargs['queryset'] = ReviewActionReason.objects.filter(
+                is_active__exact=True
+            )
+        return super().formfield_for_foreignkey(db_field, request, **kwargs)
+
+    def get_form(self, request, obj=None, **kwargs):
+        form = super(ReviewActionReasonLogAdmin, self).get_form(request, obj, **kwargs)
+        form.base_fields['reason'].widget.can_add_related = False
+        form.base_fields['reason'].widget.can_change_related = False
+        return form
+
```

Comment:
can we have some tests?  (it's not a standard ModelAdmin - there's specific code for the form field)

Reply:
tests at least for:
- the correct reasons been available for selection (excludes inactive reasons)
- that viewing an instance that used a now inactive reason still shows the original reason

Reply:
I added a test for "the correct reasons been available for selection (excludes inactive reasons)", but in trying to add a test for the second use case, I found that it does not in fact show the original reason. I have code in place limiting the source queryset for the select to only those reasons that are active, which explains why the inactive reason is excluded, but I'm not sure how to go about making it should the current reason if it is excluded. I thought maybe I could add the current reason to the queryset using `union`, but it doesn't seem like I have access to the current instance of the model object from `formfield_for_foreignkey`. Do you have any suggestions for how I might get this to work?

Reply:
As discussed on Slack, I'm now including all of the reasons in the select, but I have added a label of `(** inactive **)` to inactive reasons which hopefully should discourage admins from choosing them.